ast-checks/src/bin/missing-patch-comment.rs

```diff
@@ -56,9 +56,10 @@ fn process_patch_list(
                     let start = item.text_range().start().to_usize() as u32;
 
                     report.push(NixpkgsHammerMessage {
-                        msg: "Please add a comment on the line above, explaining the purpose of this patch.",
+                        msg: "Please add a comment on the line above, explaining the purpose of this patch.".to_string(),
                         name: "missing-patch-comment",
                         locations: vec![SourceLocation::from_byte_index(files, file_id, start)?],
+                        link: true,
```

Comment:
Maybe we should use the builder pattern `NixpkgsHammerMessage::new(name, msg, locations).with_link(false)` (defaulting to `link = true`).  Would also allow us to get rid of the manual `to_string()` casts IIRC.

Reply:
I tried this [here](https://github.com/rmcgibbo/nixpkgs-hammering/commit/92b79eeaae66caff054bf6275752940eacf66f54), but I think it's a lot more trouble than it's worth. Still haven't gotten it to compile.

Reply:
Yeah, I forgot this is usually done using another struct (it has been several years since I last worked with Rust). Looks like there is also a crate to automate this but it probably does not make sense until we have more optional fields.

https://rust-unofficial.github.io/patterns/patterns/builder.html